File: Games/Uzta/src/main/java/jokatu/game/games/uzta/UztaConfiguration.java
package jokatu.game.games.uzta;

import jokatu.components.GameComponent;
import jokatu.components.config.GameConfiguration;
import jokatu.game.games.uzta.game.UztaFactory;
import jokatu.game.games.uzta.player.UztaPlayer;
import jokatu.game.games.uzta.views.UztaViewResolverFactory;
import jokatu.game.player.PlayerFactory;
import org.jetbrains.annotations.NotNull;
import org.springframework.beans.factory.annotation.Autowired;

import static jokatu.game.games.uzta.game.Uzta.UZTA;

/**
 * @author steven
 */
@GameComponent(gameName = UZTA)
public class UztaConfiguration implements GameConfiguration {

	private final UztaFactory uztaFactory;
	private final UztaViewResolverFactory uztaViewResolverFactory;

	@Autowired
	public UztaConfiguration(UztaFactory uztaFactory,
	                         UztaViewResolverFactory uztaViewResolverFactory) {
		this.uztaFactory = uztaFactory;
		this.uztaViewResolverFactory = uztaViewResolverFactory;
	}

	@NotNull
	@Override
	public UztaFactory getGameFactory() {
		return uztaFactory;
	}

	@NotNull
	@Override
	public PlayerFactory<?> getPlayerFactory() {
		return (game, username) -> new UztaPlayer(username);
	}

	@NotNull
	@Override
	public UztaViewResolverFactory getViewResolverFactory() {
		return uztaViewResolverFactory;
	}
}

File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/UztaConfiguration.kt
package jokatu.game.games.uzta

import jokatu.components.GameComponent
import jokatu.components.config.GameConfiguration
import jokatu.game.Game
import jokatu.game.games.uzta.game.Uzta.Companion.UZTA
import jokatu.game.games.uzta.game.UztaFactory
import jokatu.game.games.uzta.player.UztaPlayer
import jokatu.game.games.uzta.views.UztaViewResolverFactory
import jokatu.game.player.PlayerFactory
import org.springframework.beans.factory.annotation.Autowired

/**
 * @author steven
 */
@GameComponent(gameName = UZTA)
class UztaConfiguration @Autowired
constructor(override val gameFactory: UztaFactory,
			override val viewResolverFactory: UztaViewResolverFactory) : GameConfiguration {

	override val playerFactory: PlayerFactory<*>
		get() = object : PlayerFactory<UztaPlayer> {
			override fun produce(game: Game<*>, username: String) = UztaPlayer(username)
		}
}
